.github/workflows/linux_simd.yml

```
@@ -67,220 +65,147 @@ jobs:
-    # ICL implies SKX, CLX and CNL
-    - name: Run SIMD tests (Ice Lake)
-      run: sde -icl -- python runtests.py -n -v -- -k test_simd
-
-  intel_spr_sde_test:
-    needs: [smoke_test]
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.4.0
-      with:
-        submodules: recursive
-        fetch-depth: 0
     - uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4.7.0
       with:
         python-version: '3.11'
+
     - name: Install Intel SDE
       run: |
         curl -o /tmp/sde.tar.xz https://downloadmirror.intel.com/784319/sde-external-9.24.0-2023-07-13-lin.tar.xz
         mkdir /tmp/sde && tar -xvf /tmp/sde.tar.xz -C /tmp/sde/
         sudo mv /tmp/sde/* /opt/sde && sudo ln -s /opt/sde/sde64 /usr/bin/sde
+
     - name: Install dependencies
       run: |
-        python -m pip install -r test_requirements.txt
-        sudo apt install gcc-12 g++-12
-    - name: Build and install NumPy
-      run: |
-        export CC=/usr/bin/gcc-12
-        export CXX=/usr/bin/g++-12
-        python setup.py develop
-    - name: Show config
+        sudo apt install -y g++-13
+        sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-13 1
+        sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-13 1
+        python -m pip install -r build_requirements.txt
+        python -m pip install pytest pytest-xdist hypothesis typing_extensions
+
+    - name: Build
+      run: spin build -- -Dallow-noblas=true -Dcpu-baseline=avx512f -Dtest-simd='BASELINE,AVX512_KNL,AVX512_KNM,AVX512_SKX,AVX512_CLX,AVX512_CNL,AVX512_ICL,AVX512_SPR'
+
+    - name: Meson Log
+      if: always()
+      run: cat build/meson-logs/meson-log.txt
+
+    - name: SIMD tests (KNM)
       run: |
-        python -c "import numpy as np; np.show_config()"
-    # Run only a few tests, running everything in an SDE takes a long time
-    # Using pytest directly, unable to use python runtests.py -n -t ...
-    - name: Run linalg/ufunc/umath tests
+        export NUMPY_SITE=$(realpath build-install/usr/lib/python*/site-packages/)
+        export PYTHONPATH="$PYTHONPATH:$NUMPY_SITE"
+        cd build-install &&
+        sde -knm -- python -c "import numpy; numpy.show_config()" &&
+        sde -knm -- python -m pytest $NUMPY_SITE/numpy/core/tests/test_simd*
+
+    - name: SIMD tests (SPR)
       run: |
-        python -m pytest numpy/core/tests/test_umath* numpy/core/tests/test_ufunc.py numpy/linalg/tests/test_*
-        # Can't run on SDE just yet: see https://github.com/numpy/numpy/issues/23545#issuecomment-1659047365
-        #sde -spr -- python -m pytest numpy/core/tests/test_umath* numpy/core/tests/test_ufunc.py numpy/linalg/tests/test_*
+        export NUMPY_SITE=$(realpath build-install/usr/lib/python*/site-packages/)
+        export PYTHONPATH="$PYTHONPATH:$NUMPY_SITE"
+        cd build-install &&
+        sde -spr -- python -c "import numpy; numpy.show_config()" &&
```

This was meant to show CPU info of the machine the CI job landed on (was useful to figure out the bug in SDE). 

```suggestion
        python -c "import numpy; numpy.show_config()" &&
```

---

The `sde -spr` usage doesn't seem needed to me here indeed. But it'll work either way, the output in the CI log seems fine. Does that sound right? If so, I can take this along in a follow-up (need to make more CI tweaks soon).